mutation/mutation_partition.cc

```diff
@@ -2505,6 +2505,7 @@ memory::reclaiming_result mutation_cleaner_impl::clear_some() noexcept {
     if (_versions.empty()) {
         return memory::reclaiming_result::reclaimed_nothing;
     }
+    auto dirty_guard = make_region_space_guard();
     auto&& alloc = current_allocator();
     partition_version& pv = _versions.front();
     if (pv.clear_gently(_tracker) == stop_iteration::yes) {
```

Comment:
We have to be super careful here. IIRC at this point the memtable's region has been joined to the cache region. So it clear_gently allocated anything, it could cause some cache eviction, which our space_guard (+2 points for the name to whoever came up with it) will account as freed memory.

Reply:
> So it clear_gently allocated anything

It mustn't. `clear_gently` is a noexcept cleanup function, it mustn't allocate.